synpurge/purger.py

```diff
@@ -39,7 +39,7 @@ def __str__(self):
         return "\n".join(lines)
 
 
-@attr.s(slots=True)
+@attr.s(slots=True, frozen=True)
```

Comment:
This PR is worth merging even if it's just for having this fix merged. Nice catch here :+1: